MDANSE_GUI/Src/MDANSE_GUI/Tabs/TrajectoryTab.py

```diff
@@ -60,16 +60,20 @@ def load_trajectories(self):
             str(self.get_path("trajectory")),
             "HDF5 files, MDANSE or H5MD format (*.mdt *.h5);;H5MD files (*.h5);;All files (*)",
         )
+
+        loaded_files = []  # list to store loaded files for recent files tracking
+
         for fname in fnames[0]:
-            self.load_trajectory(PurePath(fname))
-            last_path = str(PurePath(os.path.split(fname)[0]))
-        if fnames[0]:
-            self.set_path("trajectory", str(PurePath(last_path)))
+            filepath = Path(fname)
+            self.load_trajectory(filepath)
+            last_path = str(filepath.parent)
+            loaded_files.append(str(filepath))
+            self.set_path("trajectory", last_path)
             self._session.save()
 
     @Slot(str)
     def load_trajectory(self, some_fname: str):
-        fname = str(PurePath(some_fname))
+        fname = str(Path(some_fname))
```

Comment:
I don't understand why this line is here or what it's doing. Later it's only treated like a string, and I'm not sure the Path/str loop will do anything.